rust/capture/src/utils.rs

```diff
@@ -36,3 +38,153 @@ pub fn uuid_v7() -> Uuid {
 
     encode_unix_timestamp_millis(now_millis, &bytes)
 }
+
+pub fn replace_invalid_hex_escape_strings(
+    json_str: String,
+) -> Result<String, std::string::FromUtf8Error> {
+    // Consume the String and get its bytes
+    let mut bytes = json_str.into_bytes();
+    let len = bytes.len();
+    let mut i = 0;
+
+    const REPLACEMENT: &[u8; 4] = b"FFFD";
+    const HIGH_SURROGATE_RANGE: RangeInclusive<u16> = 0xD800..=0xDBFF;
+    const LOW_SURROGATE_RANGE: RangeInclusive<u16> = 0xDC00..=0xDFFF;
+
+    while i < len {
+        if bytes[i] == b'\\' && i + 1 < len && bytes[i + 1] == b'u' {
```

Comment:
Since this doesn't look _backward_, wouldn't this catch and replace something like `\\uD800`? (i.e. it is escaped, meaning it is literally the string `\uD800` and not the unicode codepoint)

Maybe we don't care either way, but it could deserve a test to document it as intended.

Reply:
Gah, yeah, definitely mishandled that. Fixed and added tests for that case